scripts/src/java/org/oppia/android/scripts/lint/LintProjectDescription.kt

```
@@ -0,0 +1,481 @@
+
+      if (aarFiles.isEmpty()) {
+        return emptyList()
+      }
+
+      val moduleAarsDirectory =
+        ensureDirectoryExists(File(extractedAarsDirectory, module.moduleName))
+
+      return aarFiles.asSequence()
+        .mapNotNull { aarFile -> processAarFile(aarFile, moduleAarsDirectory) }
+        .toList()
+    }
+
+    /** Resolves the JAR files for the given module. */
+    fun resolveJarFiles(module: ModuleName): List<String> {
+
+      val allDependencies = getDependenciesWithCache(module.moduleName)
+      return allDependencies.asSequence()
+        .filter { it.endsWith(".${SdkConstants.EXT_JAR}") }
+        .mapNotNull { jarFile ->
+          PathResolver.resolveBazelPath(jarFile, repoRoot, bazelClient)
+        }
+        .toList()
+    }
+
+    /** Extracts lint check JAR files from the given list of AAR files. */
+    fun extractLintCheckJars(aarFiles: List<AarFileInfo>): List<String> =
+      aarFiles.asSequence()
+        .mapNotNull { aarInfo ->
+          val lintJar = File(aarInfo.extractedPath, "lint.jar")
+          if (lintJar.exists()) lintJar.absolutePath else null
+        }
+        .toList()
+
+    private fun getDependenciesWithCache(moduleName: String): List<String> =
+      dependencyCache.getOrPut(moduleName) {
+        bazelClient.retrieveTargetModuleDependencies("//$moduleName:*")
+      }
+
+    private fun processAarFile(aarFile: String, moduleAarsDirectory: File): AarFileInfo? {
+      val resolvedAarPath = PathResolver.resolveBazelPath(aarFile, repoRoot, bazelClient)
+        ?: return null
+
+      val aarFileObj = File(resolvedAarPath)
+      require(aarFileObj.exists()) {
+        "AAR file does not exist: $resolvedAarPath"
+      }
+
+      return try {
+        val extractedPath = AarExtractor.extractAar(resolvedAarPath, moduleAarsDirectory)
+        AarFileInfo(resolvedAarPath, extractedPath)
+      } catch (e: Exception) {
+        throw IllegalStateException("Failed to extract AAR file: $aarFile", e)
```

We are swallowing all exceptions and rethrowing as `IllegalStateException`. This will lead to losing important context. Can we have more specific exception handling?

---

Updated